test/jasmine/tests/histogram_test.js

```diff
@@ -664,3 +666,207 @@ describe('Test histogram', function() {
         });
     });
 });
+
+describe('getBinSpanLabelRound', function() {
+    function _test(leftGap, rightGap, edges, calendar, expected) {
+        var ax = {type: 'not date'};
+        // only date axes have any different treatment here. We could explicitly
+        // test category
+        if(calendar) {
+            ax = {type: 'date', calendar: 'gregorian', range: [0, 1e7]};
+            setConvert(ax);
+        }
+
+        var roundFn = getBinSpanLabelRound(leftGap, rightGap, edges, ax, calendar);
+
+        var j = 0;
+        var PREC = calendar ? 1 : 6;
+        edges.forEach(function(edge, i) {
+            if(i) {
+                expect(roundFn(edge, true)).toBeCloseTo(expected[j], PREC, 'right ' + i);
+                j++;
+            }
+            if(i < edges.length - 1) {
+                expect(roundFn(edge)).toBeCloseTo(expected[j], PREC, 'left ' + i);
+                j++;
+            }
+        });
+    }
+
+    it('works when the bin edges are round numbers and data are "continuous"', function() {
+        _test(0.05, 0.3, [0, 2, 4, 6], false, [0, 1.9, 2, 3.9, 4, 5.9]);
+        _test(0, 0.1, [0, 1, 2], false, [0, 0.9, 1, 1.9]);
+        _test(0, 0.001, [0, 1, 2], false, [0, 0.999, 1, 1.999]);
+        _test(0.1, 0.1, [115, 125, 135], false, [115, 124.9, 125, 134.9]);
+        _test(0.1, 0.01, [115, 125, 135], false, [115, 124.99, 125, 134.99]);
+        _test(10, 100, [5000, 6000, 7000], false, [5000, 5900, 6000, 6900]);
+
+        // too small of a right gap: stop disambiguating data at the edge
+        _test(0, 0.0009, [0, 1, 2], false, [0, 1, 1, 2]);
+        _test(0.1, 0.009, [115, 125, 135], false, [115, 125, 125, 135]);
```

Comment:
@etpinard @chriddyp as discussed offline this morning - compare to lines 699 and 701 above that are just on the other side of the cutoff I implemented in https://github.com/plotly/plotly.js/pull/2113/commits/ad0e08f14bc18e3b8f5a570d5372db15696d86d0 - seem reasonable?

Reply:
to be clear, what these tests are indicating is that the hover labels for these bins will be `115 - 124.99` and `125 - 134.99` on the disambiguated side of the cutoff, and `115 - 125` and `125 - 135` on the too-many-digits-to-disambiguate side.